MovieTime/MovieTime/Core/MTNetworkError.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  MTNetworkError.swift
+//  MovieTime
+//
+//  Created by Enrique Melgarejo on 08/01/18.
+//  Copyright © 2018 Choynowski. All rights reserved.
+//
+
+import Foundation
+
+final class MTNetworkError: Error {
+    var statusCode: Int
+    var message: String
+
+    init(code: Int = -1, message: String? = nil) {
+        self.statusCode = code
+        self.message = message ?? "An unknown error happened."
+    }
+
+    static func unknownError() -> MTNetworkError {
+        return MTNetworkError()
+    }
+
+    static func parseError() -> MTNetworkError {
+        return MTNetworkError(code: -1001, message: "Internal server error.")
```

Comment:
what is being parsed by this method?

Reply:
When an internal error occurs while attempting to parse some data:

```
// Parse Genres
let genresParser = MTParser<MTModelMovieDBGenresList>()
let genres = try? genresParser.parse(from: data, with: genresParser.dateDecodingStrategy())
if genres != nil {
    // Save Genres
    MTApplicationCoordinator.shared.saveGenres(data: data)

    // Information Loaded
    self?.onInformationLoaded?()
} else {
    self?.onInformationFailed?(MTNetworkError.parseError().message)
}
```
The error for this is:
```
MTNetworkError(code: -1001, message: "Internal server error.")
```
The code and the message were chosen by me.

Reply:
thanks! I thought this method would parse an error, but now if I understood you correctly it creates an error object to represent an error during data parsing